internal/git/util.go

```diff
@@ -51,92 +52,129 @@ func GetLatestManifests(
 	deployablePath := localRepoPath + "/" + gitSync.Spec.Path
-	// validate the deployablePath path for its existence.
+	// validate the existence of path.
 	if _, err := os.Stat(deployablePath); err != nil {
-		return "", manifests, fmt.Errorf("invalid deployable path, err: %v", err)
+		return "", nil, fmt.Errorf("invalid repo path, err: %v", err)
 	}
 
-	// initialize the kustomize with KUSTOMIZE_BINARY_PATH if defined.
-	k := kustomize.NewKustomizeApp(deployablePath, gitSync.Spec.RepoUrl, os.Getenv("KUSTOMIZE_BINARY_PATH"))
-
-	// initialize the helm with HELM_BINARY_PATH if defined.
-	h := helm.New(deployablePath, os.Getenv("HELM_BINARY_PATH"))
-
 	sourceType, err := gitSync.Spec.ExplicitType()
 	if err != nil {
-		return "", manifests, err
+		return "", nil, err
 	}
 
+	var targetObjs []*unstructured.Unstructured
 	switch sourceType {
 	case v1alpha1.SourceTypeKustomize:
-		generatedManifest, err := k.Build(nil)
-		if err != nil {
-			logger.Errorw("can not build kustomize yaml", "err", err)
-			return "", manifests, err
+		k := kustomize.NewKustomizeApp(localRepoPath, gitSync.Spec.Path, argoGit.NopCreds{}, gitSync.Spec.RepoUrl, os.Getenv("KUSTOMIZE_BINARY_PATH"))
+		targetObjs, _, err = k.Build(nil, nil, nil)
+	case v1alpha1.SourceTypeHelm:
+		targetObjs, err = helmTemplate(localRepoPath, gitSync)
+	case v1alpha1.SourceTypeRaw:
+		targetObjs, err = rawTemplate(r, gitSync, hash)
+	}
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to build manifests, err: %v", err)
+	}
+
+	manifests := make([]string, 0)
+	for _, obj := range targetObjs {
+		if obj == nil {
+			continue
 		}
-		manifestData, err := SplitYAMLToString([]byte(generatedManifest))
+		manifestStr, err := json.Marshal(obj.Object)
 		if err != nil {
-			return "", manifests, fmt.Errorf("can not parse kustomize manifest, err: %v", err)
+			return "", nil, err
 		}
-		manifests = append(manifests, manifestData...)
-	case v1alpha1.SourceTypeHelm:
-		generatedManifest, err := h.Build(gitSync.Name, gitSync.Spec.Destination.Namespace, gitSync.Spec.Helm.Parameters, gitSync.Spec.Helm.ValueFiles)
-		if err != nil {
-			return "", manifests, fmt.Errorf("cannot build helm manifest, err: %v", err)
+		manifests = append(manifests, string(manifestStr))
+	}
+	return hash.String(), manifests, nil
+}
+
+// helmTemplate will return the list of unstructured objects after templating the helm chart.
+func helmTemplate(localRepoPath string, gitSync *v1alpha1.GitSync) ([]*unstructured.Unstructured, error) {
+	h, err := helm.NewHelmApp(localRepoPath+"/"+gitSync.Spec.Path, []helm.HelmRepository{{Repo: gitSync.Spec.RepoUrl}},
+		false, "", "", false)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing helm app object: %w", err)
+	}
+
+	templateOpts := &helm.TemplateOpts{
+		Name:      gitSync.Name,
+		Namespace: gitSync.Spec.Destination.Namespace,
+		Set:       map[string]string{},
+		SkipCrds:  true,
```

Comment:
What is this config for?

Reply:
This config is for generating the helm manifest by calling method from argo-cd helm package
`out, err := h.Template(templateOpts)`

Reply:
Sorry I should be specific, I mean `SkipCrds` this is hardcoded as `true` now.